# Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
# Use subject to license terms.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the
# "Software"), to deal in the Software without restriction, including
# without limitation the rights to use, copy, modify, merge, publish,
# distribute, and/or sell copies of the Software, and to permit persons
# to whom the Software is furnished to do so, provided that the above
# copyright notice(s) and this permission notice appear in all copies of
# the Software and that both the above copyright notice(s) and this
# permission notice appear in supporting documentation.
# 
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
# OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
# HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
# INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
# FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
# NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
# WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
# 
# Except as contained in this notice, the name of a copyright holder
# shall not be used in advertising or otherwise to promote the sale, use
# or other dealings in this Software without prior written authorization
# of the copyright holder.
#

Change needed to resume from S3 suspend on Rage XL chipset in Ultra 20.
https://bugs.freedesktop.org/show_bug.cgi?id=6829

diff -urp -x '*~' -x '*.orig' src/aticonsole.c src/aticonsole.c
--- src/aticonsole.c	2008-03-25 23:09:43.000000000 -0700
+++ src/aticonsole.c	2009-01-12 18:24:36.680149000 -0800
@@ -514,6 +514,16 @@ ATIEnterGraphics
     ATIPtr      pATI
 )
 {
+    CARD32 PciReg;
+    pciVideoPtr pVideo = pATI->PCIInfo;
+
+/*
+* Possibly fix block I/O indicator in PCI configuration space.
+*/
+    PCI_READ_LONG(pVideo, &PciReg, PCI_REG_USERCONFIG);
+    if (!(PciReg & 0x00000004U))
+	PCI_WRITE_LONG(pVideo, (PciReg | 0x00000004U), PCI_REG_USERCONFIG);
+
     /* Map apertures */
     if (!ATIMapApertures(pScreenInfo->scrnIndex, pATI))
         return FALSE;
